src/Confluent.Kafka/TopicPartitionTimestamp.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright 2016-2017 Confluent Inc., 2015-2016 Andreas Heider
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents a Kafka (topic, partition, timestamp) tuple.
+    /// </summary>
+    public class TopicPartitionTimestamp
+    {
+        /// <summary>
+        ///     Initializes a new TopicPartitionTimestamp instance.
+        /// </summary>
+        /// <param name="tp">
+        ///     Kafka topic name and partition.
+        /// </param>
+        /// <param name="timestamp">
+        ///     A Kafka timestamp value.
+        /// </param>
+        public TopicPartitionTimestamp(TopicPartition tp, Timestamp timestamp)
+            : this (tp.Topic, tp.Partition, timestamp)
+        {
+        }
+
+        /// <summary>
+        ///     Initializes a new TopicPartitionTimestamp instance.
+        /// </summary>
+        /// <param name="topic">
+        ///     A Kafka topic name.
+        /// </param>
+        /// <param name="partition">
+        ///     A Kafka partition.
+        /// </param>
+        /// <param name="timestamp">
+        ///     A Kafka timestamp value.
+        /// </param>
+        public TopicPartitionTimestamp(string topic, int partition, Timestamp timestamp)
```

Comment:
For reference, some previous conversation related to DateTime/Timestamps is in #120